plugins/destination/postgresql/client/migrate.go

```diff
@@ -2,44 +2,164 @@ package client
+	// |              3 | users       | email       | text       | NO            | true 		 | cq_users_pk         |
+	// |              1 | posts       | id          | bigint     | YES           | true 		 | cq_posts_pk			   |
+	// |              2 | posts       | title       | text       | NO            | false 	   | 					           |
+	selectAllTables = `
+SELECT
+	columns.ordinal_position AS ordinal_position,
+	pg_class.relname AS table_name,
+	pg_attribute.attname AS column_name,
+	pg_catalog.format_type(pg_attribute.atttypid, pg_attribute.atttypmod) AS data_type,
+	CASE 
+		WHEN conkey IS NOT NULL AND array_position(conkey, pg_attribute.attnum) > 0 THEN true
+		ELSE false
+	END AS is_primary_key,
+	CASE 
+		WHEN pg_attribute.attnotnull THEN true
+		ELSE false
+	END AS not_null,
+	COALESCE(pg_constraint.conname, '') AS primary_key_constraint_name
+FROM
+	pg_catalog.pg_attribute
+	INNER JOIN
+	pg_catalog.pg_class ON pg_class.oid = pg_attribute.attrelid
+	INNER JOIN
+	pg_catalog.pg_namespace ON pg_namespace.oid = pg_class.relnamespace
+	LEFT JOIN
+	pg_catalog.pg_constraint ON pg_constraint.conrelid = pg_attribute.attrelid
+	AND conkey IS NOT NULL AND array_position(conkey, pg_attribute.attnum) > 0
+	AND contype = 'p'
+	INNER JOIN
+	information_schema.columns ON columns.table_name = pg_class.relname AND columns.column_name = pg_attribute.attname AND columns.table_schema = pg_catalog.pg_namespace.nspname
+WHERE
+	pg_attribute.attnum > 0
+	AND NOT pg_attribute.attisdropped
+	AND pg_catalog.pg_namespace.nspname = '%s'
+ORDER BY
+	table_name ASC , ordinal_position ASC;
 `
 )
 
+func (c *Client) listPgTables(ctx context.Context, pluginTables schema.Tables) (schema.Tables, error) {
+	var tables schema.Tables
+	rows, err := c.conn.Query(ctx, fmt.Sprintf(selectAllTables, c.currentSchemaName))
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var ordinalPosition int
+		var tableName, columnName, columnType, pkName string
+		var isPrimaryKey, notNull bool
+		if err := rows.Scan(&ordinalPosition, &tableName, &columnName, &columnType, &isPrimaryKey, &notNull, &pkName); err != nil {
+			return nil, err
+		}
+		// This is a small optimization to hold only the tables that are in the plugin
+		if pluginTables.Get(tableName) == nil {
+			continue
+		}
+		if ordinalPosition == 1 {
+			tables = append(tables, &schema.Table{
+				Name:    tableName,
+				Columns: make([]schema.Column, 0),
+			})
+		}
+		table := tables[len(tables)-1]
+		if pkName != "" {
+			table.PkConstraintName = pkName
+		}
+		table.Columns = append(table.Columns, schema.Column{
+			Name: columnName,
+			CreationOptions: schema.ColumnCreationOptions{
+				PrimaryKey: isPrimaryKey,
+				NotNull:    notNull,
+			},
+			Type: c.PgToSchemaType(columnType),
+		})
+	}
+	return tables, nil
+}
+
+func (*Client) normalizeTableCockroach(table *schema.Table) *schema.Table {
+	for i := range table.Columns {
```

Comment:
Here we're modifying the original table spec instead of creating a copy.
This might give use some differences in the resulting data:
1. originally, column type was `cidr`
2. we've updated it to be `inet`
3. migrated (maybe even no-op)
4. writing data and marshaling it as `inet` instead of `cidr`

Reply:
cockroch doesn't support cidr hence the normalization so we can use built-in methods to compare tables.